src/pykeen/losses.py

```diff
@@ -427,20 +661,11 @@ def forward(
     if cls.synonyms is not None
     for synonym in cls.synonyms
 }
+# TODO: Can we use Resolver.from_subclasses?
```

Comment:
@cthoyt any reason why we can't? We could also set the synonym at this place.

Reply:
we could do this, I guess since we're here we might as well. I'll take care of it before we finish this PR